### PR TITLE
fix: use the keyCodes provided in options from wrapper.trigger

### PR DESCRIPTION
**Pull request by JessicaSachs**
This will address #1285 and #1295 once merged.

Previously `wrapper.trigger('keydown', { keyCode: 65 })` would trigger a keydown event with a `keyCode` of `0`. This is incorrect.

To fix this, we must listen to the keyCode if the modifier doesn't exist.

### CODE REVIEW

File: packages/test-utils/src/create-dom-event.js [hunk 1]
```diff
@@ -35,14 +72,12 @@ function createEvent(
...
+  const event = new SupportedEventInterface(
+    type,
     // event properties can only be added when the event is instantiated
     // custom properties must be added after the event has been instantiated
-    ...options,
-    bubbles,
-    cancelable,
-    keyCode: modifiers[modifier]
-  })
+    sanitizeOptions(type, modifier, { bubbles, cancelable }, options)
```
Comment (JessicaSachs) [inline packages/test-utils/src/create-dom-event.js]: Ah. I think the cheapest fix for this would be the following, which keeps feature parity and will let the modifier win over the keyCode.

```js
{
    ...options,
    bubbles,
    cancelable,
    keyCode: modifiers[modifer] || options.keyCode || options.code
}
```

File: packages/test-utils/src/create-dom-event.js [hunk 2]
```diff
@@ -24,6 +24,19 @@ const modifiers = {
   pagedown: 34
 }
 
+function getOptions(modifier, { bubbles, cancelable }, options) {
```
Comment (dobromir-hristov) [inline packages/test-utils/src/create-dom-event.js]: Maybe make the third param part of the second? No need for a third separate parameter.

File: packages/test-utils/src/create-dom-event.js [hunk 3]
```diff
@@ -24,6 +24,19 @@ const modifiers = {
...
+  const keyCode = modifiers[modifier] || options.keyCode || options.code
```
Comment (dobromir-hristov) [inline packages/test-utils/src/create-dom-event.js:29]: `code` and `keyCode` are already spread in, if provided, here we are just ensuring we dont unnecessarily overwrite then with modifiers, right?

File: packages/test-utils/src/create-dom-event.js (hunk 2 above)
Comment (JessicaSachs) [reply packages/test-utils/src/create-dom-event.js]: Keeping parity with the calling method.

File: packages/test-utils/src/create-dom-event.js (hunk 3 above)
Comment (JessicaSachs) [reply packages/test-utils/src/create-dom-event.js:29]: Yes. Previously the bug was that modifiers was falsey, yet still overwriting the keyCode.

File: packages/test-utils/src/create-dom-event.js [hunk 4]
```diff
@@ -24,6 +24,19 @@ const modifiers = {
...
+
+  const derivedOptions = { keyCode, code: keyCode }
```
Comment (afontcu) [inline packages/test-utils/src/create-dom-event.js]: I'm sure I might be missing something here, but why do we need to create the `derivedOptions` variable instead of directly adding them to the return statement?

```js
  return {
    ...options, // What the user passed in as the second argument to #trigger
    bubbles,
    cancelable,
    keyCode,
    code: keyCode
  }
```

Comment (JessicaSachs) [reply packages/test-utils/src/create-dom-event.js]: So that it’s very clear what order wins in the override/merge.  We’re gonna have more props in the future. The extra object acts as a logical grouping and communicates why certain properties win over others.

Comment (JessicaSachs) [reply packages/test-utils/src/create-dom-event.js]: I’m happy to in-line it and use a comment block as a delimiter if that’s preferred

File: packages/test-utils/src/create-dom-event.js (hunk 2 above)
Comment (JessicaSachs) [reply packages/test-utils/src/create-dom-event.js]: After fixing the failing Karma tests, I refactored the method signatures to be more compact

**Review by lmiller1990 — APPROVED** [review]
Looked over the previous comments that have been addressed, this all seems reasonable - will give it a quick test tonight (and try find edge cases) but code looks ok. Nice job!!

**Review by dobromir-hristov — APPROVED** [review]

**Review by afontcu — APPROVED** [review]
